benchopt/plotting/html/static/result.js

```diff
@@ -1,182 +1,307 @@
-      layout = graph.layout; // get layout to recover only axis
-      switch (e.value) {
-        case "loglog":
-          layout.xaxis.type = "log";
-          layout.yaxis.type = "log";
-          break;
-        case "semilog-y":
-          layout.xaxis.type = "linear";
-          layout.yaxis.type = "log";
-          break;
-        case "semilog-x":
-          layout.xaxis.type = "log";
-          layout.yaxis.type = "linear";
-          break;
-        case "linear":
-          layout.xaxis.type = "linear";
-          layout.yaxis.type = "linear";
-          break;
-      }
-      Plotly.relayout(graph.id, layout); // change axis type of plot
-    }
-  }
-}
+document.getElementById('btn_subinfo').addEventListener('click', event => {
+  const elmt = document.getElementById('subinfo');
+  const plus = document.getElementById('btn_plus');
+  const minus = document.getElementById('btn_minus');
+
+  if (elmt.style.display === 'none') {
+      elmt.style.display = 'block';
+      plus.style.display = 'none';
+      minus.style.display = 'inline';
 
-// modify the + into a - when clicking to show more system informations
-$(".toggle").click(function () {
-  $(this).find("svg").toggleClass("fa-plus-circle fa-minus-circle");
-  x = document.getElementById("subinfo");
-  if (x.style.display === "none") {
-    x.style.display = "block";
-  } else {
-    x.style.display = "none";
+      return;
   }
+
+  elmt.style.display = 'none';
+  plus.style.display = 'inline';
+  minus.style.display = 'none';
 });
 
-/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-* Toggle shades on/off on plotly graph
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
-function toggleShades() {
-  toggler = document.getElementById("change_shades");
-  if (toggler.checked === true) {
-    visible = true;
-  } else {
-    visible = false;
-  } // hide or show traces depending on toggler state
-
-  nowId = getId("now"); // id of current graph
-  graph = document.getElementById(
-    document.getElementById(nowId).getElementsByTagName("div")[1].id
-  );
-  allTraces = graph.data;
-  const allIndex = (arr) => {
-    return arr.map((elm, idx) => {
-      group = elm.legendgroup;
-      main = arr.find(function (el) { return el.legendgroup === group });
-      if ([undefined, true].includes(main.visible)) {
-        return elm.name == null ? idx : "";
-      }
-      else{ return "" }
-    }
-      ).filter(String);
-  };
-  whereToggle = allIndex(allTraces); // shade fills are without name
-  if (globalState.plot_kind[1] !== "bar_chart") {
-    Plotly.restyle(graph, { visible: visible }, whereToggle); // toggle visibility
+/*
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * Tools
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+const data = () => window.data[state().dataset][state().objective][state().objective_column];
+
+const getSolvers = () => Object.keys(data().solvers);
+
+const isBarChart = () => state().plot_kind === 'bar_chart';
```

Comment:
Can you hide the scale menu (with `loglog`) for barchart?

Reply:
Yes